server/src/test/java/com/google/fhir/gateway/BearerAuthorizationInterceptorTest.java

```diff
@@ -391,4 +395,37 @@ public String postProcess(HttpResponse response) throws IOException {
         requestDetails.getParameters().get("param3"),
         arrayContainingInAnyOrder("param3-value2", "param3-value1"));
   }
+
+  @Test
+  public void shouldSendGzippedResponseWhenRequested() throws IOException {
+    testInstance = createTestInstance(true, null);
+    String responseJson = "{\"resourceType\": \"Bundle\"}";
+    JWTCreator.Builder jwtBuilder = JWT.create().withIssuer(TOKEN_ISSUER);
+    when(requestMock.getHeader("Authorization")).thenReturn("Bearer " + signJwt(jwtBuilder));
+    when(requestMock.getHeader("Accept-Encoding".toLowerCase())).thenReturn("gzip");
+    /**
```

Comment:
nit: Is it common to have this style of comment inside a method? I usually see these as javadoc for methods/classes. I think in most places we just use `//` and maybe follow that for consistency if there is no specific reason not to.

Reply:
Changed the comments style